# disallows usage of deprecated elements (`deprecated`)

HTML5 deprecated many old elements.

## Rule details

Examples of **incorrect** code for this rule:

```html
<center>...</center>
<big>...</big>
```

Examples of **correct** code for this rule:

```html
<main>...</main>
```
