---
docType: rule
name: wcag/h30
category: a11y
summary: "WCAG H30: Providing link text"
standards:
  - wcag-2.2-a
  - wcag-2.1-a
  - wcag-2.0-a
---

# WCAG H30: Providing link text

[WCAG H30][1] requires each `<a href>` anchor link to have a text
describing the purpose of the text. The description may come from plain text or
from an image with alternative text.

The purpose of this rule is to ensure users can distinguish the link from other
links and determine whenever to follow the link or not. Assistive technology may
also present a list of links in which case the description is the only thing the
user will be presented with.

Links can be excluded from the accessibility tree with `aria-hidden="true"`.

The rule recognizes the following patterns:

- Text (non-whitespace) inside the link
- Images with `alt` text (non-whitespace)
- `aria-label` either on the link or at least one descendant.
- `aria-hidden` on the link
- `<svg>` with `<title>` or `<desc>`

[1]: https://www.w3.org/WAI/WCAG22/Techniques/html/H30

## Rule details

Examples of **incorrect** code for this rule:

<validate name="incorrect" rules="wcag/h30">
	<a href><img src="cat.gif"></a>
</validate>

Examples of **correct** code for this rule:

<validate name="correct" rules="wcag/h30">
	<a href>lorem ipsum</a>
	<a href><img src="cat.gif" alt="cat page"></a>
	<a href aria-label="lorem ipsum"></a>
	<a href aria-hidden="true"></a>
</validate>

## Version history

- %version% - Only validates `<a>` tags with the `href` attribute present.
- 7.10.0 - Handles when anchor (or a parent) is `hidden`.
- 6.6.0 - Handle text from SVG `<title>` and `<desc>`
- 0.24.0 - Rule added.
