---
docType: rule
name: wcag/h32
category: a11y
summary: "WCAG H32: Providing submit buttons"
standards:
  - wcag-2.2-a
  - wcag-2.1-a
  - wcag-2.0-a
---

# WCAG H32: Providing submit buttons

[WCAG H32][1] requires each `<form>` element to include at least one submit button in order to allow users to interact with the form in a predictable way.
For instance pressing enter in an input field to submit.

This rule checks for the presence of:

- `<button type="submit">`
- `<input type="submit">`
- `<input type="image">`

Submit buttons can either be nested or associated using the `form` attribute.

[1]: https://www.w3.org/WAI/WCAG22/Techniques/html/H32

## Rule details

Examples of **incorrect** code for this rule:

<validate name="incorrect" rules="wcag/h32">
    <form>
        <label>
            Text field: <input type="text">
        </label>
    </form>
</validate>

Examples of **correct** code for this rule:

<validate name="correct" rules="wcag/h32">
    <form>
        <label>
            Text field: <input type="text">
        </label>
        <button type="submit">Submit</button>
    </form>
</validate>

Submit buttons may also use the `form` attribute to associate with a form:

<validate name="associated" rules="wcag/h32">
    <form id="my-form">
        ...
    </form>
    <button form="my-form" type="submit">Submit</button>
</validate>
