---
docType: rule
name: wcag/h32
category: a17y
summary: "WCAG 2.1 H32: Providing submit buttons"
---

# WCAG 2.1 H32: Providing submit buttons (`wcag/h32`)

[WCAG 2.1 technique H32][1] requires each `<form>` element to include at least
one submit button in order to allow users to interact with the form in a
predictable way. For instance pressing enter in an input field to submit.

[1]: https://www.w3.org/WAI/WCAG21/Techniques/html/H32

## Rule details

Examples of **incorrect** code for this rule:

<validate name="incorrect" rules="wcag/h32">
    <form>
        <label>
            Text field: <input type="text">
        </label>
    </form>
</validate>

Examples of **correct** code for this rule:

<validate name="correct" rules="wcag/h32">
    <form>
        <label>
            Text field: <input type="text">
        </label>
        <button type="submit">Submit</button>
    </form>
</validate>
