---
docType: rule
name: wcag/h36
category: a11y
summary: "WCAG H36: Require alt text on images used as submit buttons"
standards:
  - wcag-2.2-a
  - wcag-2.1-a
  - wcag-2.0-a
---

# WCAG H36: Require alt text on images used as submit buttons

[WCAG H36][1] requires all images used as submit buttons to have a
textual description using the `alt` attribute. The text must indicate the
buttons function but not describe the image.

[1]: https://www.w3.org/WAI/WCAG22/Techniques/html/H36

## Rule details

Examples of **incorrect** code for this rule:

<validate name="incorrect" rules="wcag/h36">
	<a><img src="cat.gif"></a>
</validate>

Examples of **correct** code for this rule:

<validate name="correct" rules="wcag/h36">
	<a>lorem ipsum</a>
	<a><img src="cat.gif" alt="cat page"></a>
</validate>
