---
docType: rule
name: wcag/h36
category: a17y
summary: "WCAG 2.1 H36: Require alt text on images used as submit buttons"
---

# WCAG 2.1 H36: Require alt text on images used as submit buttons (`wcag/h36`)

[WCAG 2.1 technique H36][1] requires all images used as submit buttons to have a
textual description using the `alt` attribute. The text must indicate the
buttons function but not describe the image.

[1]: https://www.w3.org/WAI/WCAG21/Techniques/html/H36

## Rule details

Examples of **incorrect** code for this rule:

<validate name="incorrect" rules="wcag/h36">
	<a><img src="cat.gif"></a>
</validate>

Examples of **correct** code for this rule:

<validate name="correct" rules="wcag/h36">
	<a>lorem ipsum</a>
	<a><img src="cat.gif" alt="cat page"></a>
</validate>
