---
docType: rule
name: wcag/h36
category: a11y
summary: "WCAG H36: Require alt text on images used as submit buttons"
standards:
  - wcag-2.2-a
  - wcag-2.1-a
  - wcag-2.0-a
---

# WCAG H36: Require alt text on images used as submit buttons

[WCAG H36][wcag/h36] requires all images used as submit buttons to have a non-empty textual description using the `alt` attribute.
The text must indicate the buttons function but not describe the image.

[wcag/h36]: https://www.w3.org/WAI/WCAG22/Techniques/html/H36

## Rule details

Examples of **incorrect** code for this rule:

<validate name="incorrect" rules="wcag/h36">
<!-- missing alt attribute -->
<input type="image" src="submit-button.png">

<!-- empty alt attribute -->
<input type="image" src="submit-button.png" alt="">
</validate>

Examples of **correct** code for this rule:

<validate name="correct" rules="wcag/h36">
	<input type="image" src="submit-button.png" alt="Submit Button">
</validate>
