---
docType: rule
name: wcag/h37
category: a11y
summary: "WCAG H37: Using alt attributes on img elements"
standards:
  - wcag-2.2-a
  - wcag-2.1-a
  - wcag-2.0-a
---

# Require images to have alternative text

Both [HTML5][1] and [WCAG][2] requires all images to have an alternative text. Assistive technology such as screenreaders relies on the alt-text to provide a description of the image.

Purely Decorative images must use empty (`null`) `alt=""` attribute (preferred) or be excluded from accessibility tree with `role="presentation"` or `aria-hidden="true"`.

In addition to `<img>` this rules also checks for alternative text on:

- `<input type="image">`

[1]: https://html.spec.whatwg.org/#alt
[2]: https://www.w3.org/WAI/WCAG22/Techniques/html/H37

## Rule details

Examples of **incorrect** code for this rule:

<validate name="incorrect" rules="wcag/h37">
    <img>
    <input type="image">
</validate>

Examples of **correct** code for this rule:

<validate name="correct" rules="wcag/h37">
    <img alt="...">
    <input type="image" alt="..">
</validate>

## Options

This rule takes an optional object:

```json
{
  "allowEmpty": true,
  "alias": []
}
```

### `allowEmpty`

Sometimes images are used in context where the image only adds to the user experience but is already clear from an A11Y perspective.

When `true` this allows empty `<img alt="">` to be used to prevent errors and to signal that the image is not meant to be read.

Examples of **correct** code with this option enabled:

<validate name="allow-empty" rules="wcag/h37" wcag/h37='{"allowEmpty": true}'>
<span>The task was successfully completed! <img src="thumbsup.png" alt=""></span>
</validate>

### `alias`

If javascript is used to set the attribute on-the-fly (e.g. using `angular-translate`) the alternative attributes can be listed here.

Examples of **correct** code when `alias` is `["data-alt"]`:

<validate name="alias" rules="wcag/h37" wcag/h37='{"alias": ["data-alt"]}'>
<img data-alt="...">
</validate>
