---
docType: rule
name: wcag/h63
category: a11y
summary: "WCAG H63: Using the scope attribute to associate header cells and data cells"
standards:
  - wcag-2.2-a
  - wcag-2.1-a
  - wcag-2.0-a
---

# WCAG H63: Using the scope attribute to associate header cells and data cells

[WCAG H63][1] requires header cells to have scope to associate with data cells.

[1]: https://www.w3.org/WAI/WCAG22/Techniques/html/H63

## Rule details

Examples of **incorrect** code for this rule:

<validate name="incorrect-missing" rules="wcag/h63">
    <th></th>
</validate>

<validate name="incorrect-auto" rules="wcag/h63">
    <!-- auto state cannot be set with a value -->
    <th scope="auto"></th>
</validate>

Examples of **correct** code for this rule:

<validate name="correct-row" rules="wcag/h63">
    <th scope="row"></th>
</validate>

<validate name="correct-col" rules="wcag/h63">
    <th scope="col"></th>
</validate>

<validate name="correct-rowgroup" rules="wcag/h63">
    <th scope="rowgroup"></th>
</validate>

<validate name="correct-colgroup" rules="wcag/h63">
    <th scope="colgroup"></th>
</validate>

## Markdown

If you are using content generated from Markdown consider disabling this rule as most markdown parsers does not generate `<th>` elements with the scope attribute.

Alternatively use a {@link usage#inline-configuration disable directive} somewhere in the markdown file:

```md
<!-- [html-validate-disable-block wcag/h63: markdown does not generate tables with scope attribute] -->

| Foo | Bar | Baz |
| --- | --- | --- |
| 1   | 2   | 3   |
```

## Version history

- 7.10.0 - Rule added.
