@ngdoc rule
@name wcag/h67
@category a17y
@summary WCAG 2.1 H67: Using null alt text and no title attribute on img elements for images that AT should ignore
@description

# WCAG 2.1 H67: Using null alt text and no title attribute on img elements for images that AT should ignore (`wcag/h67`)

[WCAG 2.1 technique H67][1] requires decorative images to have empty alt text
(`alt=""`) and empty or absent `title`.

[1]: https://www.w3.org/WAI/WCAG21/Techniques/html/H67

## Rule details

Examples of **incorrect** code for this rule:

<validate name="incorrect" rules="wcag/h67">
    <img title="Lorem ipsum">
</validate>

Examples of **correct** code for this rule:

<validate name="correct" rules="wcag/h67">
    <!-- empty alt text and no title is interpretated as purely decorative -->
    <img alt="">

    <!-- alt text is used together with title, the image carries meaning and will not be ignored by AT -->
    <img alt="Lorem ipsum" title="Lorem ipsum">

</validate>
