---
docType: rule
name: wcag/h71
category: a11y
summary: "WCAG H71: Providing a description for groups of form controls"
standards:
  - wcag-2.2-a
  - wcag-2.1-a
  - wcag-2.0-a
---

# WCAG H71: Providing a description for groups of form controls

[WCAG H71][1] requires all fieldsets to have a `<legend>` element as first child element.

[1]: https://www.w3.org/WAI/WCAG22/Techniques/html/H71

## Rule details

Examples of **incorrect** code for this rule:

<validate name="incorrect" rules="wcag/h71">
	<fieldset>
		...
	</fieldset>
</validate>

Examples of **correct** code for this rule:

<validate name="correct" rules="wcag/h71">
	<fieldset>
		<legend>Lorem ipsum</legend>
			...
	</fieldset>
</validate>
