---
docType: rule
name: wcag/h71
category: a17y
summary: "WCAG 2.1 H71: Providing a description for groups of form controls"
---

# WCAG 2.1 H71: Providing a description for groups of form controls (`wcag/h71`)

[WCAG 2.1 technique H71][1] requires all fieldsets to have a `<legend>` element as first child element.

[1]: https://www.w3.org/WAI/WCAG21/Techniques/html/H71

## Rule details

Examples of **incorrect** code for this rule:

<validate name="incorrect" rules="wcag/h71">
	<fieldset>
		...
	</fieldset>
</validate>

Examples of **correct** code for this rule:

<validate name="correct" rules="wcag/h71">
	<fieldset>
		<legend>Lorem ipsum</legend>
			...
	</fieldset>
</validate>
